src/uu/seq/src/seq.rs

```diff
@@ -286,16 +286,15 @@ fn print_seq(
     let (first, increment, last) = range;
     let mut i = 0isize;
     let mut value = first + i as f64 * increment;
+    let width = if pad { padding + largest_dec } else { 0 };
```

Comment:
I think you need:
```suggestion
    let width = if pad { padding + 1 + largest_dec } else { 0 };
```
Can you add a test case for `seq -w 9.0 10.0`? That'd catch this, it's supposed to output `"09.0\n10.0\n"`.

Reply:
Yes, I'll add that.

Reply:
It appears it's not as simple as this. Interestingly, this change causes the following failures:
```
failures:

---- test_seq::test_equalize_widths_floats stdout ----
current_directory_resolved: 
run: /home/jeffrey/src/coreutils/target/debug/coreutils seq -w 5 10.0
thread 'test_seq::test_equalize_widths_floats' panicked at 'assertion failed: `(left == right)`

Diff < left / right > :
<"005\n006\n007\n008\n009\n010\n"
>"05\n06\n07\n08\n09\n10\n"

', tests/common/util.rs:229:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- test_seq::test_width_scientific_notation stdout ----
current_directory_resolved: 
run: /home/jeffrey/src/coreutils/target/debug/coreutils seq -w 999 1e3
thread 'test_seq::test_width_scientific_notation' panicked at 'assertion failed: `(left == right)`

Diff < left / right > :
<"00999\n01000\n"
>"0999\n1000\n"

', tests/common/util.rs:229:9
```
This seems to be happening because `seq -w 999 1e3`, for example, is using the `print_seq()` code path, not the `print_seq_integers()` code path, since the `last` argument is a float, and the `print_seq_integers()` code path is only executed when all the arguments---`first`, `last`, and `increment`---are `BigInt`s.

However, it really doesn't matter that the `last` argument is a float except at the time of comparison in the `done_printing()` function. In other words, the `last` argument shouldn't matter when deciding whether to call `print_seq()` or `print_seq_integers()`, and it shouldn't matter when computing the width.

Reply:
I added an additional commit to this branch that I hope will handle this issue.